lib/CHI/Driver/DBI.pm

```diff
@@ -0,0 +1,313 @@
+package CHI::Driver::DBI;
+
+use strict;
+use warnings;
+
+use DBI::Const::GetInfoType;
+use Moose;
+use Moose::Util::TypeConstraints;
+use Carp qw(croak);
+use Time::HiRes qw(time);
+
+our $VERSION = '1.24';
+
+extends 'CHI::Driver';
+
+my $type = "CHI::Driver::DBI";
+
+subtype "$type.DBIHandleGenerator" => as 'CodeRef';
+subtype "$type.DBIXConnector"      => as 'DBIx::Connector';
+subtype "$type.DBIHandle"          => as 'DBI::db';
+
+coerce "$type.DBIHandleGenerator" => from "$type.DBIXConnector" => via {
+    my $dbixconn = $_;
+    sub { $dbixconn->dbh }
+};
+coerce "$type.DBIHandleGenerator" => from "$type.DBIHandle" => via {
+    my $dbh = $_;
+    sub { $dbh }
+};
+
+has 'db_name'      => ( is => 'rw', isa => 'Str' );
+has 'dbh'          => ( is => 'ro', isa => "$type.DBIHandleGenerator", coerce => 1 );
+has 'dbh_ro'       => ( is => 'ro', isa => "$type.DBIHandleGenerator", predicate => 'has_dbh_ro', coerce => 1 );
+has 'sql_strings'  => ( is => 'rw', isa => 'HashRef', lazy_build => 1 );
+has 'table' => ( is => 'rw', isa => 'Str', default => 'chi_cache' );
+has 'key_prefix' => ( is => 'rw', isa => 'Str', builder => '_build_key_prefix', lazy => 1);
+
+__PACKAGE__->meta->make_immutable;
+
+sub BUILD {
+    my ( $self, $args ) = @_;
+
+    my $dbh = $self->dbh->();
+
+    $self->db_name( $dbh->get_info( $GetInfoType{SQL_DBMS_NAME} ) );
+    $self->sql_strings;
+
+    if ( $args->{create_table} ) {
+        $dbh->do( $self->sql_strings->{create} )
+          or croak $dbh->errstr;
+    }
+
+    return;
+}
+
+sub _table {
+    my ( $self, ) = @_;
+
+    return $self->table();
+}
+
+sub _build_key_prefix {
+    my ( $self ) = @_;
+    return $self->namespace . "::";
+}
+
+sub namespaced_key {
+    my ($self, $key) = @_;
+    return $self->key_prefix . $key;
+}
+
+sub _build_sql_strings {
+    my ( $self, ) = @_;
+
+    my $dbh   = $self->dbh->();
+    my $table = $dbh->quote_identifier( $self->_table );
+    my $value = $dbh->quote_identifier('value');
+    my $expires_at = $dbh->quote_identifier('expires_at');
+    my $key_prefix_match = $dbh->quote( $self->key_prefix . "%" );
+    my $key   =  $dbh->quote_identifier('key');
+
+    my $strings = {
+        fetch    => "SELECT $value FROM $table WHERE $key = ? and ? < $expires_at",
+        store    => "INSERT INTO $table ( $key, $value, $expires_at ) VALUES ( ?, ?, ? )",
+        store2   => "UPDATE $table SET $value = ?, $expires_at = ? WHERE $key = ?",
+        remove   => "DELETE FROM $table WHERE $key = ?",
+        clear    => "DELETE FROM $table where $key like $key_prefix_match",
+        get_keys => "SELECT DISTINCT $key FROM $table where $key like $key_prefix_match and ? < $expires_at",
+        create   => "CREATE TABLE IF NOT EXISTS $table ("
+          . " $key VARCHAR( 300 ), $value TEXT, $expires_at REAL,"
```

Comment:
I fixed it without the KEY since SQLite3 doesn't support it and the driver tests are based on it